core/src/main/java/org/apache/carbondata/core/datastore/page/SafeVarLengthColumnPage.java

```diff
@@ -40,6 +40,7 @@
   @Override
   public void freeMemory() {
     byteArrayData = null;
+    rowOffset.freeMemory();
```

Comment:
rowOffset.freeMemory also required in SafeDecimalColumnPage.
It will we good practice to call rowOffset.freeMemory  in same class it is allocated, call abstract freememory class from derived to ensure any clean up from abstract class

Reply:
ok